lib/matplotlib/colors.py

```diff
@@ -1220,6 +1215,24 @@ def _init(self):
         self._isinit = True
         self._set_extremes()
 
+    @property
+    def monochrome(self):
+        """Return whether all colors in the colormap are identical."""
+        # Replacement for the attribute *monochrome*. This ensures a consistent
+        # response independent of the way the ListedColormap was created, which
+        # was not the case for the manually set attribute.
+        #
+        # TODO: It's a separate discussion whether we need this property on
+        #       colormaps at all (at least as public API). It's a very special edge
+        #       case and we only use it for contours internally.
+        if not self._isinit:
+            self._init()
+
+        N = len(self._lut) - 3  # disregard over / under / bad
+        if N <= 1:
+            return True
+        return np.all(self._lut[0] == self._lut[1:N])
```

Comment:
Can you use the `self.N` attribute here?
```suggestion
        return self.N <=1 or np.all(self._lut[0] == self._lut[1:self.N])
```

Reply:
Yes, for now. I ultimately want to do something about self.N (remove/make private/ turn into property - t.b.d.) but until then, we may as well use it.